test/quic_client_test.c

```diff
@@ -157,20 +193,34 @@ static int test_quic_client(void)
     SSL_CTX_free(c_ctx);
     BIO_ADDR_free(s_addr_);
     BIO_free(c_net_bio_own);
-    if (c_fd != INVALID_SOCKET)
+    if (cc_fd == INVALID_SOCKET && c_fd != INVALID_SOCKET)
         BIO_closesocket(c_fd);
     return testresult;
 }
 
-OPT_TEST_DECLARE_USAGE("certfile privkeyfile\n")
+OPT_TEST_DECLARE_USAGE("certfile connect_first privkeyfile\n")
 
 int setup_tests(void)
 {
+    char *mode;
+
     if (!test_skip_common_options()) {
         TEST_error("Error parsing test options\n");
         return 0;
     }
 
+    if (test_get_argument_count() > 0) {
+	mode = test_get_argument(0);
+	if (TEST_ptr(mode) && strcasecmp(mode, "connect_first") == 0) {
+	    connect_socket();
+	}
+
+	if (cc_fd == INVALID_SOCKET) {
+	    TEST_error("Could not create connected socket");
+	    return 0;
+	}
+    }
```

Comment:
Not sure if it is worth changing it (probably not) but it would be much easier to just make the `test_quic_client` to have 2 iterations. (`ADD_ALL_TESTS(test_quic_client, 2);`) In one iteration you would use a connected socket and in the other unconnected one.